File: examples/typescript/src/order.ts
export type OrderState = "Draft" | "Confirmed";

/** @aidd.requirement urn:aidd:order:requirement:confirmed-cannot-return-draft */
export function confirm(state: OrderState): OrderState {
  if (state !== "Draft") {
    throw new Error("Only draft orders can be confirmed");
  }
  return "Confirmed";
}

File: examples/kotlin/src/main/kotlin/Order.kt
package example

enum class OrderState {
    DRAFT,
    CONFIRMED,
}

/**
 * @aidd.requirement urn:aidd:order:requirement:confirmed-cannot-return-draft
 */
fun confirm(state: OrderState): OrderState {
    require(state == OrderState.DRAFT)
    return OrderState.CONFIRMED
}

